src/System.Console/src/System/Console.cs

```diff
@@ -25,33 +25,16 @@ public static class Console
         private static ConsoleCancelEventHandler _cancelCallbacks;
         private static ConsolePal.ControlCHandlerRegistrar _registrar;
 
-        internal static T EnsureInitialized<T>(ref T field, Func<T> initializer) where T : class
-        {
-            lock (InternalSyncObject)
-            {
-                T result = Volatile.Read(ref field);
-                if (result == null)
-                {
-                    result = initializer();
-                    Volatile.Write(ref field, result);
-                }
-                return result;
-            }
-        }
+        internal static T EnsureInitializedWrapper<T>(ref T field, Func<T> initializer) where T : class =>
```

Comment:
Is that ok? Because otherwise, I will have to spill out InternalSyncObject in multiple places where EnsureInitialized(Wrapper) has been used.